examples/trl/sft.py

```diff
@@ -164,6 +167,12 @@ def create_datasets(tokenizer, args, seed=None):
     base_model.generation_config.flash_attention_recompute = script_args.flash_attention_recompute
     base_model.generation_config.flash_attention_causal_mask = script_args.flash_attention_causal_mask
 
+    apply_zero3_leaf_promotion(
+        base_model,
+        is_deepspeed_zero3_enabled=is_zero3_enabled,
+        use_zero3_leaf_promotion=training_args.use_zero3_leaf_promotion,
+    )
```

Comment:
what do you think of `if is_zero3_enabled and training_args.use_zero3_leaf_promotion: apply_zero3_leaf_promotion(base_model)`
where the method apply_zero3_leaf_promotion is only be called when it's functionality is needed